GUI/src/pages/Training/Intents/CommonIntents.tsx

```diff
@@ -77,20 +76,20 @@ const CommonIntents: FC = () => {
       };
       commonIntents.push(newIntent);
     });
+    intentParam = searchParams.get('intent');
   }
 
   useEffect(() => {
-    const queryIntentName = searchParams.get('intent');
-    if (commonIntents && queryIntentName) {
-      const queryIntent = commonIntents.find(
-        (intent) => intent.intent === queryIntentName
-      );
-      if (queryIntent) {
-        setSelectedIntent(queryIntent);
-        setSelectedTab(queryIntentName);
-      }
+    if (!intentParam || intentsFullList?.length !== commonIntents?.length) return;
+
+    const queryIntent = commonIntents.find(
+      (intent) => intent.id === intentParam
+    );
+
+    if (queryIntent) {
+      setSelectedIntent(queryIntent);
     }
-  }, [commonIntents, searchParams]);
+  }, [intentParam]);
```

Comment:
Can intentsFullList and commonIntents change? If so useEffect is not executed, is it intentional?

Reply:
Discussed, all intentional.